src/App.jsx

```diff
@@ -0,0 +1,48 @@
+import './App.css';
+import {
+  BrowserRouter as Router,
+  Route,
+  Routes,
+  Navigate,
+} from 'react-router-dom';
+import Header from './features/Header/components/Header.jsx';
+import ProductPage from './features/ProductPage/components/ProductPage.jsx';
+import ProductInfoPage from './features/ProductInfoPage/components/ProductInfoPage';
+import CartPage from './features/CartPage/components/CartPage';
+import React, {useEffect} from 'react';
+import {setCart} from './features/CartPage/slices/cartSlice';
+import {useDispatch} from 'react-redux';
+import Snackbar from './common/components/Snackbar';
+import CheckoutPage from './features/CheckoutPage/components/CheckoutPage';
+import OrderPlacedPage from './features/OrderPlacedPage/components/OrderPlacedPage';
+import OrdersPage from './features/OrdersPage/components/OrdersPage';
+
+const App = () => {
+  const dispatch = useDispatch();
+  useEffect(()=>{
```

Comment:
for life Cycle hooks, the FE best practice is to create a custom hook and export all lifecycle hooks from there. 

eg:
https://github.com/browserstack/frontend/blob/master/apps/boiler-plate/src/features/Counter/components/useCounter.js

please do this for all lyfecycle hooks in the application.

Reply:
do you want me to remove all `useEffect` 😨 ? That's a very major change.

Reply:
That's how our prod applications are made, you'll be working on that kindof codebase. hence the assignment essentially tests adherence to such workspace rules. hence it should also cover that. it's an industry best-practice.

Reply:
Understandable, Please keep in mind for FE tasks.